src/pybind/cephfs/setup.py

```diff
@@ -1,56 +1,196 @@
-            self.egg_base = build_command.build_base
-            self.egg_info = os.path.join(self.egg_base, os.path.basename(self.egg_info))
+            cflags['extras'].append(cflag)
+
+    for ldflag in subprocess.check_output(
+            [python_config, "--ldflags"]
+    ).strip().decode('utf-8').split():
+        if ldflag.startswith('-l'):
+            ldflags['l'].append(ldflag.replace('-l', ''))
+        if ldflag.startswith('-L'):
+            ldflags['L'].append(ldflag.replace('-L', ''))
+        else:
+            ldflags['extras'].append(ldflag)
+
+    return {
+        'cflags': cflags,
+        'ldflags': ldflags
+    }
+
+
+def check_sanity():
+    """
+    Test if development headers and library for cephfs is available by compiling a dummy C program.
+    """
+
+    tmp_dir = tempfile.mkdtemp(dir=os.environ.get('TMPDIR', os.path.dirname(__file__)))
+    tmp_file = os.path.join(tmp_dir, 'cephfs_dummy.c')
+
+    with open(tmp_file, 'w') as fp:
+        dummy_prog = textwrap.dedent("""
+        #include <stddef.h>
+        #include <cephfs/libcephfs.h>
+
+        int main(void) {
+            struct ceph_mount_info *cmount = NULL;
+            ceph_init(cmount);
+            return 0;
+        }
+        """)
+        fp.write(dummy_prog)
+
+    compiler = new_compiler()
+    customize_compiler(compiler)
+
+    if {'MAKEFLAGS', 'MFLAGS', 'MAKELEVEL'}.issubset(set(os.environ.keys())):
+        # The setup.py has been invoked by a top-level Ceph make.
+        # Set the appropriate CFLAGS and LDFLAGS
+        CEPH_SRC_DIR = os.path.join(
+            os.path.dirname(os.path.abspath(__file__)),
+            '..',
+            '..'
+        )
+
+        compiler.set_include_dirs(os.path.join(CEPH_SRC_DIR, 'include'))
+        compiler.add_library_dir(os.environ.get('CEPH_LIBDIR'))
+
+    try:
+        compiler.define_macro('_FILE_OFFSET_BITS', '64')
+
+        compiler.link_executable(
+            compiler.compile([tmp_file], tmp_dir),
+            os.path.join(tmp_dir, 'cephfs_dummy'),
+            libraries=['cephfs', 'rados'],
+            output_dir=tmp_dir,
+        )
+
+    except CompileError:
+        print('\nCompile Error: Ceph FS development headers not found', file=sys.stderr)
+        return False
+    except LinkError:
+        print('\nLink Error: Ceph FS library not found', file=sys.stderr)
+        return False
+    else:
+        return True
+    finally:
+        shutil.rmtree(tmp_dir)
+
+
+if not check_sanity():
+    sys.exit(1)
+
+cmdclass = {}
+try:
+    from Cython.Build import cythonize
+    from Cython.Distutils import build_ext
+
+    cmdclass = {'build_ext': build_ext}
+except ImportError:
+    print("WARNING: Cython is not installed.")
+
+    if not os.path.isfile('cephfs.c'):
+        print('ERROR: Cannot find Cythonized file cephfs.c', file=sys.stderr)
+        sys.exit(1)
+    else:
+        def cythonize(x, **kwargs):
+            return x
+
```

Comment:
please remove at least one blank line here.

and could you run this script thru pep8 to kill some warnings.